Add release notes for 4.3.0

alhambrav opened the pull request:
### Ticket reference or full description of what's in the PR
Add release notes for 4.3.0

alhambrav committed:
Start release notes for 4.2.3

alhambrav committed:
Update support page

alhambrav commented:
Note added for https://github.com/craftercms/craftercms/issues/7689

alhambrav committed:
Merge branch 'master' into rn-4-2-3

alhambrav committed:
Update release notes for #7699

alhambrav commented:
Note added for https://github.com/craftercms/craftercms/issues/7699

sumerjabri reviewed (changes requested):
Switch to 4.3. Add Box removal as a breaking change.

alhambrav committed:
Merge branch 'master' into rn-4-2-3

alhambrav committed:
Updates for 4.3.0 release

alhambrav committed:
Updates for box #7728

alhambrav commented:
Added box removal as breaking change and in release notes commit:  https://github.com/craftercms/docs/pull/1944/commits/9f40578ed6f1184b2f4f36a8c24fb1a8994d1181 
Updated box mentions in commit: https://github.com/craftercms/docs/pull/1944/commits/446f89c320a7147177ab87c6778efa94559496cb

Related to https://github.com/craftercms/craftercms/issues/7728

alhambrav committed:
Update description for #7699

alhambrav committed:
Update version in version picker

sumerjabri commented on source/release-notes/4-3-0.rst:
```diff
@@ -0,0 +1,68 @@
+^^^^^^^^^^^^^^^^^^
+Noteworthy dependency updates:
+
+- OpenSearch to version 2.20
```
I don't think that's true. We're still on OpenSearch 2.17.

These dep updates should be removed because there are many more that I updated, which are not visible. An upgrade of OpenSearch or MariaDB is big enough to matter. Upgrades of internal libraries like these are numerous and not worth enumerating at this time.

sumerjabri commented on source/release-notes/4-3-0.rst:
```diff
@@ -0,0 +1,68 @@
+
+* `Fixed issue with move operations on items and folders <https://github.com/craftercms/craftercms/issues/7729>`__
+
+* Fixed issue with syncing users in groups for externally managed users (IdP).
```
This seems unexplained since there is no ticket. We changed syncing such that users are synced based on IdP only for externally managed groups, but local group membership is unchanged.

alhambrav committed:
Updates based on review

alhambrav committed:
Merge branch 'master' into rn-4-2-3

alhambrav committed:
Update release notes for #7767, #7768 and #7769

alhambrav commented:
Added tickets  https://github.com/craftercms/craftercms/issues/7767, https://github.com/craftercms/craftercms/issues/7768 and https://github.com/craftercms/craftercms/issues/7769 with commit:https://github.com/craftercms/docs/pull/1944/commits/65f4753bd16ff664d60ab613027c03782174e3fc

alhambrav committed:
Fix link to API in remove_remote permission

alhambrav committed:
Merge branch 'master' into rn-4-2-3

alhambrav committed:
Merge branch 'master' into rn-4-2-3

alhambrav committed:
Update dates
